sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -289,7 +308,8 @@ class ParquetFileFormat
         // Instead, we use FileScanRDD's task completion listener to close this iterator.
         val iter = new RecordReaderIterator(vectorizedReader)
         try {
-          vectorizedReader.initialize(split, hadoopAttemptContext, Option.apply(fileFooter))
+          vectorizedReader.initialize(
+            split, hadoopAttemptContext, Some(inputFile), Some(inputStream), Some(fileFooter))
```

Comment:
nit: Although `fileFooter` should not be null, it's still advisable to use `Option(fileFooter)` just to be safe

Reply:
addressed